src/ja/worker/config.py

```diff
@@ -1,36 +1,68 @@
 from typing import Dict
-
 from ja.common.config import Config
 from ja.common.proxy.ssh import SSHConfig
 from ja.common.work_machine import ResourceAllocation
+import yaml
+from typing import cast
 
 
 class WorkerConfig(Config):
     """
     Config for the worker client.
     """
 
+    def __init__(self, conf_path: str):
+        """!
+        reads the config file from the disk
+        creates resource allocation instance
+        """
+        with open(conf_path, 'r') as stream:
+            data_loaded = yaml.safe_load(stream)
+
+        self._uid = data_loaded["uid"]
+        hostname: str = data_loaded["hostname"]
+        username: str = data_loaded["username"]
+        self._ssh_config = SSHConfig(hostname, username)
+        cpu_threads: int = cast(int, data_loaded["cpu_threads"])
+        available_memory: int = cast(int, data_loaded["available_memory"])
+        swap_memory: int = cast(int, data_loaded["swap_memory"])
+        self._resource_allocation = ResourceAllocation(cpu_threads, available_memory, swap_memory)
```

Comment:
I think you misunderstood what I meant.
You should read in properties via from_dict, not via the constructor.
The YAML file should be opened in the main class of the worker, not in WorkerConfig.